src/runtime_src/core/tools/xbmgmt2/OO_LoadConfig.cpp

```diff
@@ -126,5 +128,44 @@ OO_LoadConfig::execute(const SubCmdOptions& _options) const
     return;
   }
 
-  //TO_DO: parse the INI file
-}
\ No newline at end of file
+  try {
+    for (auto& t : deviceCollection)  
+      loadconfig(t);
+  } catch (const std::runtime_error& e) {
+    // Catch only the exceptions that we have generated earlier
+    std::cerr << boost::format("ERROR: %s\n") % e.what();
+    return;
+  }
+}
+
+/*
+ * so far, we only support the following configs
+ * [Device]
+ * mailbox_channel_disable = xxx
+ * mailbox_channel_switch = xxx
+ * cahce_xclbin = xxx
+ */
+void
+OO_LoadConfig::loadconfig(const std::shared_ptr<xrt_core::device>& _dev) const
+{
+  boost::property_tree::ptree m_tree;
+  boost::property_tree::ini_parser::read_ini(m_path, m_tree);

```

Comment:
can this throw an exception if the format isn't correct? We should have a catch in that case and reroute the error to cerr

Reply:
the function call is within a try clause, so an exception will be thrown, and err msg will be displayed